pymeasure/adapters/serial.py

```diff
@@ -78,5 +78,31 @@ def binary_values(self, command, header_bytes=0, dtype=np.float32):
         header, data = binary[:header_bytes], binary[header_bytes:]
         return np.fromstring(data, dtype=dtype)
 
+    def format_binary_values(self, values, datatype='f', is_big_endian=False):
+        """Format values in binary format in order to be used with instrument commands.
+
+        :param values: data to be writen to the device.
+        :param datatype: the format string for a single element. See struct module.
+        :param is_big_endian: boolean indicating endianess.
+        :return: binary string.
+        :rtype: bytes
+        """
+
+        block = to_ieee_block(values, datatype, is_big_endian)
+
+        return block
+
+    def write_binary_values(self, command, values, **kwargs):
+        """ Write binary data to the instrument, e.g. waveform for signal generators
+
+        :param command: SCPI command to be sent to the instrument
+        :param values: iterable representing the binary values
+        :param kwargs: Key-word arguments to pass onto `write_binary_values`
+        :returns: number of bytes written
+        """
+
+        block = self.format_binary_values(values, **kwargs)
```

Comment:
Initially, I did not have the `format_binary_values`, but for the motivations described in https://github.com/pymeasure/pymeasure/pull/408#discussion_r584095442, I think having this internal API (with leading `_`) is probably better. In addition to this, the SCPI instruments understand ieee format (e.g. prefix by #, etc. see http://ena.support.keysight.com/e5061b/manuals/webhelp/eng/programming/remote_control/reading-writing_measurement_data/data_transfer_format.htm). Indeed, the implementation provided does not support HP format (`to_hp_block`) and pure binary format (`to_binary_block`) as in the original pyvisa write_binary_values API. This could be added, but I wanted to limit the dependency on function from `pyvisa`. What do you think ?

Reply:
I think in case users have instruments expecting a different format, it would be valuable (and not too complicated) to have this user-configurable (and with equivalent API) as in [pyvisa write_binary_values](https://github.com/pyvisa/pyvisa/blob/9cab3576fb6b7c4aa18538d4337c67f8d971fcc7/pyvisa/resources/messagebased.py#L307-L314), with an iee default.
I think if we pull in one pyvisa util function, we might also pull in the other equivalent ones.

Reply:
Ok